orientdb/src/main/java/com/yahoo/ycsb/db/OrientDBClient.java

```diff
@@ -68,42 +76,70 @@ public void init() throws DBException {
-        if (newdb) {
-          System.out.println("OrientDB drop and recreate fresh db");
-          db.drop();
-          db.create();
-        }
-      } else {
-        System.out.println("OrientDB database not found, create fresh db");
-        db.create();
-      }
-
-      System.out.println("OrientDB connection created with " + url);
+    System.err.println("OrientDB loading database url = " + url);
 
-      dictionary = db.getMetadata().getIndexManager().getDictionary();
-      if (!db.getMetadata().getSchema().existsClass(CLASS))
-        db.getMetadata().getSchema().createClass(CLASS);
+    // If using a remote database, use the OServerAdmin interface to connect
+    if (url.startsWith(OEngineRemote.NAME)) {
+      isRemote = true;
+      if (remoteStorageType == null) {
+        throw new DBException("When connecting to a remote OrientDB instance, specify a database storage type (plocal or memory) with " + STORAGE_TYPE_PROPERTY);
+      }
 
-      db.declareIntent(new OIntentMassiveInsert());
+      try {
+        OServerAdmin server = new OServerAdmin(url).connect(user, password);
+        String dbpath = url.split(":", 2)[1];
+
+        if (server.existsDatabase()) {
+          if (newdb) {
+            System.err.println("OrientDB dropping and recreating fresh db on remote server.");
+            server.dropDatabase(remoteStorageType);
+            server.createDatabase(dbpath, ORIENTDB_DOCUMENT_TYPE, remoteStorageType);
+          }
+        } else {
+          System.err.println("OrientDB database not found, creating fresh db");
+          server.createDatabase(dbpath, ORIENTDB_DOCUMENT_TYPE, remoteStorageType);
+        }
 
-    } catch (Exception e1) {
-      System.err.println("Could not initialize OrientDB connection pool for Loader: " + e1.toString());
-      e1.printStackTrace();
-      return;
+        server.close();
+        db = new ODatabaseDocumentTx(url).open(user, password);
+      } catch (IOException | OException e) {
+        System.err.println(String.format("Error interfacing with %s", url));
+        e.printStackTrace();
+      }
+    } else {
+      try {
+        db = new ODatabaseDocumentTx(url);
+        if (db.exists()) {
+          db.open(user, password);
+          if (newdb) {
+            System.err.println("OrientDB dropping and recreating fresh db.");
+            db.drop();
+            db.create();
+          }
+        } else {
+          System.err.println("OrientDB database not found, creating fresh db");
+          db.create();
+        }
+      } catch (Exception e1) {
+        System.err.println("Could not initialize OrientDB connection pool for Loader: " + e1.toString());
+        e1.printStackTrace();
+        return;
+      }
     }
 
+    System.err.println("OrientDB connection created with " + url);
+    dictionary = db.getMetadata().getIndexManager().getDictionary();
+    if (!db.getMetadata().getSchema().existsClass(CLASS))
+      db.getMetadata().getSchema().createClass(CLASS);
```

Comment:
Purely a formatting change. Although this is marked as an addition, it is original code. Don't worry, it'll get fixed with the checkstyle PR :smile:.